aiohttp/client.py

```diff
@@ -444,11 +445,12 @@ async def _request(
                     if proxy is not None:
                         proxy = URL(proxy)
                     elif self._trust_env:
-                        for scheme, proxy_info in proxies_from_env().items():
-                            if scheme == url.scheme:
-                                proxy = proxy_info.proxy
-                                proxy_auth = proxy_info.proxy_auth
-                                break
+                        if not proxy_bypass(str(url.host)):
```

Comment:
FYI `url.host` is already a string.

Reply:
Removed the cast.

Reply:
Then the function typings should be adjusted I guess... FWIW I've shown the guard expression in one of the comments so it's accounted for.

Reply:
The type error says that it is `Optional[str]`, in other words, it could be `None`. I don't think `str(None)` would be the right approach to resolving the typing error.

It suggests you should probably have an `if url.host is None` check or a sensible default value with `url.host or "default"`.

Reply:
Exactly, I've incorporated this idea in the most recent bunch of the review comments.